src/qcodes/dataset/sqlite/connection.py

```diff
@@ -7,22 +7,29 @@
 from __future__ import annotations
 
 import logging
+import sqlite3
 from contextlib import contextmanager
 from typing import TYPE_CHECKING, Any
 
 import wrapt  # type: ignore[import-untyped]
+from typing_extensions import deprecated
 
-from qcodes.utils import DelayedKeyboardInterrupt
+from qcodes.utils import DelayedKeyboardInterrupt, QCoDeSDeprecationWarning
 
 if TYPE_CHECKING:
-    import sqlite3
     from collections.abc import Iterator
 
 log = logging.getLogger(__name__)
 
 
+@deprecated(
+    "ConnectionPlus is deprecated. Please use connect to create an AtomicConnection.",
+    category=QCoDeSDeprecationWarning,
+)
 class ConnectionPlus(wrapt.ObjectProxy):  # pyright: ignore[reportUntypedBaseClass]
     """
+    Note this is a legacy class. Please refer to ``AtomicConnection``
```

Comment:
for fancier docs with clickable links? :)
```suggestion
    Note this is a legacy class. Please refer to :class:`AtomicConnection`
```